File: Go/basic2/string.go
package main

import "fmt"


func removerCaractere(str string, index int) string {
    return str[:index] + str[index+1:]
}



func main () {
	
	var nome string = "ggabriel"
	
	fmt.Println(string(nome[0])) // ver um caracter de uma string
    fmt.Println("String original:", nome)

    // Removendo o primeiro caractere
   // novoNome := removerCaractere(nome, 0)
   novoNome := nome[1:]

    fmt.Println("String modificada:", novoNome)

}

File: C/string.c
#include <stdio.h>
#include <string.h>

void removerCaractere(char *str, int index) {
    int i;
    int tamanho = strlen(str);

    // Desloca os caracteres à direita do índice um para a esquerda
    for (i = index; i < tamanho - 1; i++) {
        str[i] = str[i + 1];
    }

    // Adiciona o caractere nulo ao final da nova string
    str[tamanho - 1] = '\0';
}

int main() {
    char nome[] = "ggabriel";
    printf("String original: %s\n", nome);

    // Removendo o primeiro caractere
    removerCaractere(nome, 0);

    printf("String modificada: %s\n", nome);

    return 0;
}
